### ISSUE TITLE
New "Both Hands" description

### ISSUE THREAD

**Issue by zepumph** [phetsims/ratio-and-proportion#207]
A new walkthrough in the design doc is ready for prototyping in the sim. A few key pieces:

* It uses the "closer/farther" paradigm when direction changes (like alerts for single hand interaction)
* It moves much of the context response into the object response.
* It implements new absolute position regions. These are different from the single hand interaction ones, but we will keep both for now.

**Comment by zepumph** [comment]
The basic structure of the object/context reponses is in place. I still need to update the both-hands position regions. 

The main complexity was in adding the "direction changed" text when closer/farther changes, and making that work well with the both hands and individual interaction, because both use that logic.

- [ ] I left a couple of TODOs in the sim that I want to go over, potentially with @terracoda, as they pertain to when the lock ratio checkbox is enabled. First though we should go one step at a time (hence a checkbox here so we don't forget to come back to it).

Next, new regions.

**Comment by zepumph** [comment]
New regions are implemented for qualitative, "both-hands" position. @terracoda I think this is ready for some poking around on master to see what you think. 
https://phet-dev.colorado.edu/html/ratio-and-proportion/1.0.0-dev.68/phet/

- [ ] Alerts don't repeat if they are the same sometimes. @terracoda can you reproduce this, is it problematic to you?

**Comment by terracoda** [comment]
@zepumph, when both the hands are in the same region, we can greatly reduce the length of the object response by saying, "both hands in {{region-name}}"

You might not have seen [Table: (New Idea) Parameters for Region of Play Area](https://docs.google.com/document/d/1HJvvGWYaUv9tlat9S7omQQVng8fGB4FuUhTEM1qx1z8/edit#heading=h.c9wem8ke1pju)
in the design doc. I have simplified it here for you.

**Note** that, I just added a row for being "at middle". That wasn't clear in the table yesterday. You can leave that single-value region off if it is hard to implement.

Play Area Regions | Approximate range for regions | Examples when hands are in the same region | Examples when hands are not in the same region
-- | -- | -- | --
{{at top}} | 10 | Both hands at top | Left hand at top, Right hand in upper region
{{in upper region}} | 7.1 - 9.99 | Both hands in upper region | Left hand in upper, Right hand in middle region
{{in middle region}} | 3.01- 4.49 & 5.01 - 7.00 | Both hands in middle region | Left hand in middle region, Right hand in lower region
{{at middle}} | = 5.0 | Both hands at middle | Left hand at middle, Right hand in lower region
{{in lower region}} | 0.01 - 3.00 | Both hands in lower region | Left hand in lower region, Right hand at bottom
{{at bottom}} | 0 | Both hands at bottom | Left hand at bottom, Right hand in middle region

**Comment by terracoda** [comment]
Also for distance, I don't hear "even with other hand". I'll check the design doc, maybe we haven't we requested that.

**Comment by terracoda** [comment]
@zepumph, I'm not sure the "closer/farther" paradigm when direction changes is working here because when I go back and forth from moving first the left hand up and then the right hand up, I get directional changes each time. I get closer with left hand and then farther with right hand.

If I keep doing that, I never hear actual distance :-)

So, can you you do the following:
- remove the closer/farther responses for both hands
- fix up the response strings to use Both hands {{at/in region-name}} when they are in the same region
- can you look at response pile-ups? I am getting a lot of them.

**Comment by zepumph** [comment]
>@zepumph, when both the hands are in the same region, we can greatly reduce the length of the object response by saying, "both hands in {{region-name}}"

Implemented above like "boths hands {{inPosition}}"

**Comment by zepumph** [comment]
"at middle" added above.

**Comment by zepumph** [comment]
>I don't hear "even with other hand".

This may have been a rounding issue fixed in https://github.com/phetsims/ratio-and-proportion/commit/e84cd52f71f011db3d6b806eae3578fef3864c83. I think that you figured this out in https://github.com/phetsims/ratio-and-proportion/issues/207, but I wanted to note that I can get "even with" distance when using individual interactions now, but now the both hands has been updated as you specified above. 

The majority of the complexity when originally implementing this was about the direction changes. Thus I'm a bit nervous to totally get rid of it from the code. I added a TODO to clean it up if review this and confirm that we don't want it. 

Everything has been implemented. Let me know how things sound here. Anything else?

**Comment by terracoda** [comment]
@zepumph, the combined response "Both hands {{region-name}}" is working when I use the W and S keys for the left hand, but not when I use the arrow keys for the right hand still get separate descriptions for each hand.

**Comment by terracoda** [comment]
Generally related to this issue, I posted #214. It is difficult to evaluate the quality (and length) of the responses with the large pile-ups.

**Comment by zepumph** [comment]
>"Both hands {{region-name}}" is working when I use the W and S keys for the left hand, but not when I use the arrow keys for the right hand still get separate descriptions for each hand.

Hmm, I can't reproduce this. Can you potentially be more specific? Is it a pile up error? Is it consistently in a case that you could describe? Is the error in VoiceOver, such that you see different things between the aria-live section of the a11y view and what is in your VO dialog?

**Comment by zepumph** [comment]
My main question to discuss in this issue tomorrow at the meeting is to confirm that direction description will not be needed in the both-hands interaction. That will be a sizeable cleanup that I would like to do while this work is fresh.

**Comment by zepumph** [comment]
From design meeting today, @terracoda repeated her sentiment from above comments that the direction changes won't work with the both hands interaction. As a result currently `HandPositionsDescriber.getBothHandsDistanceOrDirection()` is unused. @terracoda mentioned though that this may be an important piece of describing the lock ratio. So I have nothing else in regards to this issue.

@terracoda please let me know how to produce the left/right hand differentiation described inhttps://github.com/phetsims/ratio-and-proportion/issues/207#issuecomment-712385721.

**Comment by zepumph** [comment]
Actually, @terracoda made me aware of a large issue in the way I implemented both hands. Basically, whenever both hands are in the same region, that is when we shorten the object description. I mistook that for when the VALUES were the same, which is only at one spot. That should be fixed, and instead should say "even with" when the values are the same.

Reassigning.

**Comment by zepumph** [comment]
Description has been fixed above. Now https://github.com/phetsims/ratio-and-proportion/issues/207#issuecomment-713050690 applies. Ready for review.
